duckduckgo/iOS
# fix incorrect header names / file names and add script to check them

Task/Issue URL:
Tech Design URL:
CC:

**Description**:

Adds a run script build phase to check file headers.

Also abstracts `BGTask` in to something that can be mocked.

**Steps to test this PR**:
1. Check out the code
1. Edit one or more of the file names it the comment header so it doesn't match the file name
1. Build the code -> Should product warnings for each file that doesn't match
1. Ensure code builds and runs


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
